File: android/app/src/main/java/com/vaultsync/domain/usecase/DetectDuplicatesUseCase.kt
package com.vaultsync.domain.usecase

import com.vaultsync.domain.model.Document
import com.vaultsync.domain.repository.DocumentRepository

class DetectDuplicatesUseCase(
    private val documentRepository: DocumentRepository
) {
    suspend operator fun invoke(sha256: String): Document? {
        return documentRepository.findBySha256(sha256)
    }
}

File: ios/Sources/VaultSync/Domain/UseCases/DetectDuplicatesUseCase.swift
import Foundation

public final class DetectDuplicatesUseCase: Sendable {
    private let documentRepository: any DocumentRepository

    public init(documentRepository: any DocumentRepository) {
        self.documentRepository = documentRepository
    }

    public func execute(sha256: String) async throws -> Document? {
        try await documentRepository.findBySha256(sha256)
    }
}
